src/main/java/ch/uzh/soprafs22/groupmatcher/service/EmailService.java

```diff
@@ -59,6 +61,29 @@ public void sendResponseVerificationEmail(Student student) {
         mailSender.send(composeMessage("Verify Response", "email_verification.html", variables, student.getEmail()));
```

Comment:
To make the inline images show up change composeMessage like this:

```
private MimeMessage composeMessage(String subject, String templateName, Map<String, Object> variables, String... recipients) {
    MimeMessage message = mailSender.createMimeMessage();
    try {
        MimeMessageHelper helper = new MimeMessageHelper(message,
                MimeMessageHelper.MULTIPART_MODE_MIXED, StandardCharsets.UTF_8.name());
        Context context = new Context();
        context.setVariables(variables);
        helper.setFrom("notify@groupmatcher.ch");
        helper.setTo(recipients);
        helper.setSubject(subject);
        String html = templateEngine.process(templateName, context);
        helper.setText(html, true);
        helper.addInline("bg.png", new ClassPathResource("bg.png"));
        helper.addInline("logo.png", new ClassPathResource("logo.png"));
        helper.addInline("tagline.png", new ClassPathResource("tagline.png"));
    } catch (MessagingException exception) {
        log.info("Failed to create an email message");
    }
    return message;
}
```

And make sure the png files are under resources/, not inside templates/.